drivers/sensor/st/lsm6dso/lsm6dso_trigger.c

```diff
@@ -175,36 +412,62 @@ static void lsm6dso_handle_interrupt(const struct device *dev)
 	struct lsm6dso_data *lsm6dso = dev->data;
 	const struct lsm6dso_config *cfg = dev->config;
 	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
-	lsm6dso_status_reg_t status;
+	lsm6dso_all_sources_t sources;
+	bool pending_status = false;
 
 	while (1) {
-		if (lsm6dso_status_reg_get(ctx, &status) < 0) {
-			LOG_DBG("failed reading status reg");
+
+		if (lsm6dso_all_sources_get(ctx, &sources) < 0) {
```

Comment:
THis routine is really expensive in terms of bus cycles. I think that maybe we should avoid calling it if TAP and TILT are not enabled... What do you think?

Reply:
Yes I agree. I'll make some tests. Unfortunately tap and tilt are not available via lsm6dso_status_reg_get (it's how ST's driver is made). I will split it so that if tap and tilt are not enabled it will run the existing code accessing lsm6dso_status_reg_get.

Reply:
I split the code in two distinct handlers depending if tilt or tap are used. It seemed tricky to me to have a single function combining the two. Let me know what you think.

Reply:
Good to know. Thanks.